plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseConfig.java

```diff
@@ -13,16 +13,28 @@
  */
 package io.trino.plugin.clickhouse;
 
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableSet;
 import io.airlift.configuration.Config;
 import io.airlift.configuration.ConfigDescription;
 import io.airlift.configuration.DefunctConfig;
 
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.base.Strings.nullToEmpty;
+
 @DefunctConfig("clickhouse.legacy-driver")
 public class ClickHouseConfig
 {
     // TODO (https://github.com/trinodb/trino/issues/7102) reconsider default behavior
     private boolean mapStringAsVarchar;
 
+    private Set<String> jdbcTypesMappedToVarchar = ImmutableSet.of();
```

Comment:
Revert unrelated changes.

Reply:
ClickhouseClient construct needs the action of ClickHouseConfig.getJdbcTypesMappedToVarchar , while ClickHouseConfig  don't extend BaseJdbcConfig

Reply:
You can inject both ClickHouseConfig & BaseJdbcConfig.